run.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+# run.sh - Runs everything necessary to generate our API artifacts (for
+# website, external consumers, etc.) based on our inputs (from
+# covid-data-public).
+
+set -o nounset
+set -o errexit
+
+DEFAULT_DATA_SOURCES_DIR="../covid-data-public"
+DEFAULT_API_RESULTS_DIR="api_results"
+
+main () {
+  # Go to repo root (where run.sh lives).
+  cd "$(dirname "$0")"
+
+  # Parse args if specified.
+  if [ $# -eq 0 ]; then
+    DATA_SOURCES_DIR="$(absPath "${DEFAULT_DATA_SOURCES_DIR}")"
+    API_OUTPUT_DIR="$(absPath "${DEFAULT_API_RESULTS_DIR}")"
+  elif [ $# -ne 2 ]; then
+    echo "Usage: $0 [covid-data-public directory] [output-directory]"
+    exit 1
+  else
+    DATA_SOURCES_DIR="$(absPath $1)"
+    API_OUTPUT_DIR="$(absPath $2)"
+  fi
+
+  echo "Generating all API artifacts."
+  echo "Data sources directory: ${DATA_SOURCES_DIR}"
+  echo "API output directory: ${API_OUTPUT_DIR}"
+
+  if [ ! -d "${DATA_SOURCES_DIR}" ] ; then
+    echo "Directory ${DATA_SOURCES_DIR} does not exist."
+    exit 1
+  fi
+
+  if [ ! -d "${API_OUTPUT_DIR}" ] ; then
+    echo "Directory ${API_OUTPUT_DIR} does not exist. Creating."
+    mkdir -p "${API_OUTPUT_DIR}"
+  fi
+
+  # run_model.py uses the COVID_DATA_PUBLIC environment variable to find inputs.
+  export COVID_DATA_PUBLIC="${DATA_SOURCES_DIR}"
+
+  # These directiories essentially define the structure of our API endpoints.
+  # TODO: These should perhaps live in python, near the schemas (defined in api/)?
+  STATES_DIR="${API_OUTPUT_DIR}/";
+  # TODO: I think deploy_dod_dataset.py may currently have an implicit
+  # requirement that the county model JSON is in a /county subdirectory of the
+  # states?
+  COUNTIES_DIR="${API_OUTPUT_DIR}/county";
+  COUNTY_SUMMARIES_DIR="${API_OUTPUT_DIR}/county_summaries";
+  CASE_SUMMARIES_DIR="${API_OUTPUT_DIR}/case_summary"
+
+  # TODO: Move DoD onto a more general-purpose schema rather than treat them custom.
+  DOD_DIR="${API_OUTPUT_DIR}/custom1"
+
+  echo ">>> Generating state models to ${STATES_DIR}"
```

Comment:
it seems like up to this point, we're just setting vars.  could we just make that as part of a `prepare()` function?